docs/pbc/all/dynamic-multistore/202404.0/base-shop/how-to-add-new-custom-field.md

```diff
@@ -0,0 +1,319 @@
+
+<details>
+<summary>src/Pyz/Zed/StoreContextGui/Communication/Form/StoreContextForm.php</summary>
+
+```php
+
+namespace Pyz\Zed\StoreContextGui\Communication\Form;
+
+use Spryker\Zed\StoreContextGui\Communication\Form\StoreContextForm as SprykerStoreContextForm;
+use Symfony\Component\Form\Extension\Core\Type\EmailType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Validator\Constraints\Email;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+class StoreContextForm extends SprykerStoreContextForm
+{
+    /**
+     * @var string
+     */
+    protected const FIELD_SUPPORT_CONTACT_EMAIL = 'contactEmail';
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array<string, mixed> $options
+     *
+     * @return void
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        parent::buildForm($builder, $options);
+
+        $this->addContactEmailField($builder, $options);
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array<string, mixed> $options
+     *
+     * @return \Spryker\Zed\StoreContextGui\Communication\Form\StoreContextForm
+     */
+    protected function addContactEmailField(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add(static::FIELD_SUPPORT_CONTACT_EMAIL, EmailType::class, [
+                'label' => 'Contact Email',
+                'constraints' => $this->createEmailConstraints(),
+            ]);
+
+        return $this;
+    }
+
+    /**
+     * @return array
+     */
+    protected function createEmailConstraints(): array
+    {
+        return  [
+            new Email(),
+        ];
+    }
+}
+
+```
+
+</details>
+
+6. Replace `StoreContextForm` in the extended `StoreContextGuiCommunicationFactory`:
+
+*src/Pyz/Zed/StoreContextGui/Communication/StoreContextGuiCommunicationFactory.php*
+
+```php
+
+namespace Pyz\Zed\StoreContextGui\Communication;
+
+use Pyz\Zed\StoreContextGui\Communication\Form\StoreContextForm;
+use Spryker\Zed\StoreContextGui\Communication\StoreContextGuiCommunicationFactory as SprykerStoreContextGuiCommunicationFactory;
+
+class StoreContextGuiCommunicationFactory extends SprykerStoreContextGuiCommunicationFactory
+{
+    /**
+     * @return string
+     */
+    public function getStoreContextFormClass(): string
+    {
+        return StoreContextForm::class;
+    }
+}
+```
+
+The new field should now be available in the Back Office.
+
+
+{% info_block warningBox "Verification" %}
+
+1. In the Back Office, go to **Administration**>**Stores**.
+2. On the **Stores** page, click **Edit store** next to a store.
+3. On the **Edit store** page, click the **Settings** tab.
+In the **Settings** tab, make sure the following applies:
+  * The **Contact Email** field is displayed.
+  * The block with the **APPLICATION** value set to **Default** is displayed.
```

Comment:
Is this block added as part of this doc or why do we mention this?

Reply:
Adjusted text